runtime/src/main/java/org/corfudb/runtime/view/LayoutView.java

```diff
@@ -295,11 +295,12 @@ public void committed(long epoch, Layout layout, boolean force)
 
         int timeouts = 0;
         int responses = 0;
-        while (responses < commitList.length) {
-            // wait for someone to complete.
+
+        for (CompletableFuture cf : commitList) {
             try {
-                CFUtils.getUninterruptibly(CompletableFuture.anyOf(commitList),
-                        WrongEpochException.class, TimeoutException.class, NetworkException.class);
+                CFUtils.getUninterruptibly(cf, WrongEpochException.class,
```

Comment:
remove CFUtils usage and check other uses of anyOf in this file.

Reply:
This seems to have the issue that it will not break out to check the next future till the current one in the loop is complete.
This make things timeout if the nodes early in the loop are slow

Reply:
Are you referring to committed or other methods?

The wait time is bounded by the RPC timeout, not RPC timeout * number of requests, since the requests are pipelined

Reply:
The reason i bring this up is that the protocol does not need a success from all nodes for committed. This is more stringent than what is needed for correctness.

Reply:
This change doesn't change the semantics, it just fixes a logic bug.

We wait on all futures to complete, not on all servers to respond. The router will timeout requests, so the max time is bounded by the RPC timeout of a single request.